code/game/objects/effects/mines.dm

```diff
@@ -19,87 +40,205 @@
+	qdel(s)
+	qdel(src)
+
+/obj/effect/mine/frag/explode()
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread()
+	triggered = 1
+	s.set_up(3, 1, src)
+	s.start()
+	var/turf/O = get_turf(src)
+	if(!O)
+		return
+	src.fragmentate(O, 20, 7, list(/obj/item/projectile/bullet/pellet/fragment)) //only 20 weak fragments because you're stepping directly on it
+	qdel(s)
+	qdel(src)
 
-/obj/effect/mine/proc/explode(obj)
-	explosion(loc, 0, 1, 2, 3)
-	spawn(0)
-		qdel(src)
 
 /obj/effect/mine/dnascramble
-	name = "Radiation Mine"
+	name = "radiation mine"
+	desc = "A small explosive mine with a radiation symbol on the side."
 	icon_state = "uglymine"
-	triggerproc = "triggerrad"
+	mineitemtype = /obj/item/weapon/mine/dnascramble
 
 /obj/effect/mine/phoron
-	name = "Phoron Mine"
+	name = "incendiary mine"
+	desc = "A small explosive mine with a fire symbol on the side."
 	icon_state = "uglymine"
-	triggerproc = "triggerphoron"
+	mineitemtype = /obj/item/weapon/mine/phoron
 
 /obj/effect/mine/kick
-	name = "Kick Mine"
+	name = "kick mine"
+	desc = "Concentrated war crimes. Handle with care."
 	icon_state = "uglymine"
-	triggerproc = "triggerkick"
+	mineitemtype = /obj/item/weapon/mine/kick
 
 /obj/effect/mine/n2o
-	name = "N2O Mine"
+	name = "nitrous oxide mine"
+	desc = "A small explosive mine with three Z's on the side."
 	icon_state = "uglymine"
-	triggerproc = "triggern2o"
+	mineitemtype = /obj/item/weapon/mine/n2o
 
 /obj/effect/mine/stun
-	name = "Stun Mine"
+	name = "stun mine"
+	desc = "A small explosive mine with a lightning bolt symbol on the side."
+	icon_state = "uglymine"
+	mineitemtype = /obj/item/weapon/mine/stun
+
+/obj/effect/mine/frag
+	name = "fragmentation mine"
+	desc = "A small explosive mine with 'FRAG' and a grenade symbol on the side."
+	icon_state = "uglymine"
+	mineitemtype = /obj/item/weapon/mine/frag
+	var/fragment_types = list(/obj/item/projectile/bullet/pellet/fragment)
+	var/num_fragments = 20  //total number of fragments produced by the grenade
+	//The radius of the circle used to launch projectiles. Lower values mean less projectiles are used but if set too low gaps may appear in the spread pattern
+	var/spread_range = 7
+
+/obj/item/weapon/mine
+	name = "mine"
+	desc = "A small explosive mine with 'HE' and a grenade symbol on the side."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "uglymine"
+	var/arming = 0
+	var/countdown = 10
+	var/minetype = /obj/effect/mine
+
+/obj/item/weapon/mine/attack_self(mob/user as mob)
+	if(!arming)
+		to_chat(user, "<span class='warning'>You prime \the [name]! [countdown] seconds!</span>")
+		activate(user)
+		add_fingerprint(user)
+		if(iscarbon(user))
+			var/mob/living/carbon/C = user
+			C.throw_mode_on()
+	else
+		to_chat(user, "You cancel \the [name]'s priming sequence.")
+		arming = 0
+		countdown = initial(countdown)
+		icon_state = initial(icon_state)
+		add_fingerprint(user)
+	return
+
+/obj/item/weapon/mine/proc/activate(mob/user as mob)

```

Comment:
This whole proc could probably be done better as a do_after() in the attack_self().